plugins/woocommerce/client/admin/client/marketplace/components/product-card/product-card.tsx

```diff
@@ -108,29 +112,46 @@ function ProductCard( props: ProductCardProps ): JSX.Element {
 		queueRecordEvent( event, data );
 	}
 
-	const isTheme = type === ProductType.theme;
-	const isBusinessService = type === ProductType.businessService;
-	let productVendor: string | JSX.Element | null = product?.vendorName;
-	if ( ! isCompact && product?.vendorName && product?.vendorUrl ) {
-		productVendor = (
+	const createVendorLink = ( eventName: string ) => {
+		if ( ! product?.vendorName || ! product?.vendorUrl ) {
+			return product?.vendorName || null;
+		}
+
+		return (
 			<a
 				href={ product.vendorUrl }
 				rel="noopener noreferrer"
+				target="_blank"
 				onClick={ () => {
-					recordTracksEvent(
-						'marketplace_product_card_vendor_clicked',
-						{
-							product: product.title,
-							vendor: product.vendorName,
-							product_type: type,
-						}
-					);
+					recordTracksEvent( eventName, {
+						product: product.title,
+						vendor: product.vendorName,
+						product_type: type,
+					} );
 				} }
 			>
 				{ product.vendorName }
```

Comment:
Screen reader text for opening the link in a new tab is being introduced via #57771